### ISSUE TITLE
Pre 0.7.0 testing

### ISSUE THREAD

**Issue by plexus** [plexus/chestnut#74]
I'd like to release a new stable version as soon as Om 0.8.0 is out. it's at rc1 now so this might be soon. Before we can release though we'll have to verify everything still works as expected.

A description of what "basic test" and "production test" means can be found [in the docs](http://plexus.github.io/chestnut/testing.html).

* [x] Basic Test ; no options
* [x] Basic Test ; --http-kit 
* [x] Basic Test ; --site-middleware
* [x] Basic Test ; --om-tools
* [ ] Basic Test ; --cljx 
* [x] Basic Test ; --less 
* [x] Basic Test ; --speclj
* [x] Basic Test ; all options
* [ ] Production Test ; no options
* [ ] Production Test ; --http-kit 
* [ ] Production Test ; --site-middleware
* [ ] Production Test ; --om-tools
* [ ] Production Test ; --cljx 
* [ ] Production Test ; --less 
* [ ] Production Test ; --speclj
* [ ] Production Test ; all options
* [ ] Feature Test : CLJX
* [ ] Feature Test : Speclj

Known issues that need to be fixed before 0.7.0 : 

* [ ] [LESS broken on Heroku](https://github.com/plexus/chestnut/issues/72)

**Comment by asok** [comment]
Hey, I could help out with manual testing. I could take on the "Basic Test". One question though - how can someone upgrade chestnut using leiningen?

**Comment by plexus** [comment]
Good question... what I usually do is checkout the source, and then from the root directory do `lein install`. I'm not sure what happens though if we release 0.7.0 and people still have 0.6.0 in their local maven repository. Will leiningen automatically get the latest version? no idea.

**Comment by asok** [comment]
To make sure I use the correct version of chestnut I've first removed the old one and then I did `lein install`. But it seems I cannot use the new version of chestnut. My terminal session:
```sh
$ rm -rf ~/.m2/repository/chestnut
$ lein install
Created /Users/asok/Documents/projects/chestnut/target/lein-template-0.7.0-SNAPSHOT.jar
Wrote /Users/asok/Documents/projects/chestnut/pom.xml
Installed jar and pom into local repo.
$ cd ../
$ lein new chestnut foo
Failed to resolve version for chestnut:lein-template:jar:RELEASE
This could be due to a typo in :dependencies or network issues.
If you are behind a proxy, try setting the 'http_proxy' environment variable.
Could not find template chestnut on the classpath.
```

I'm a clojure newbie really, any help?

**Comment by plexus** [comment]
Hey Adam,

No problem. The reason it doesn't find the version you've installed is
because you've installed a snapshot version, and leiningen is looking for a
stable version.

lein new chestnut --snapshot

should do the trick.

**Comment by plexus** [comment]
FWIW I'm doing some experimenting with automated testing in the `test` branch. The idea is to use [expect4j](https://github.com/cverges/expect4j) and Selenium. There's still a lot to figure out, so might be a while before something useful emerges, but I've done too much of this manual testing in the past already.

**Comment by plexus** [comment]
First test has landed, see #89 . Leaving this open just a little longer for reference, we'll have to implement all the above test cases, but no longer needed to do it manually.
